V1Michigan/v1
# Profile page functionality

### Status
**HOLD until #7 is ready** -- let's merge that first, then I'll fix the merge conflicts

### Description
Basic profile page functionality -- view your + other users' profiles, edit your profile
- Remove old `/account` page
- Add `/profile/[username]`, which shows [username]'s profile, and allows editing all fields of their profile if `currentUsername === [username]`
- Add `/profile`, which redirects to `/profile/[currentUsername]`
- Move onboarding form elements from `Step1` and `Step2` into `components/profile/fields/` so that they can be re-used for the `/profile` page (this also lets us re-use validation logic :smiley:)
- Replace TypeScript enums for Year, RoleType, and FieldOfStudy with regular objects...TS enums were really painful to work with, and we weren't getting much value from type-checking these

~Still TODO, I will get these done tonight:~
- ~Handling profile data fetching errors~ :heavy_check_mark: 
- ~Not sure why GH Actions build is failing :dizzy_face:~ :heavy_check_mark: 

### Dependencies
- ADDED react-pdf (had to change Next Webpack config to make sure it doesn't try to render PDFs with SSR, see https://github.com/V1Michigan/v1/pull/8/commits/816b0684e3ae62b7cdd34157b646de3f8003ac96)

### Migrations
NO

### Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

```sh
git pull --prune
git checkout profile-page
npm run dev
```

1. Delete your old user (from the Supabase console)
2. Go through the sign-up + onboarding flow again -- should look the same, but there have been some changes under the hood, make sure validation still works as expected
3. Go to `/profile` to see your profile
4. Edit your profile by clicking "edit profile" at the bottom of the `/profile/[username]` page
5. Confirm that your profile (including avatar + resume) was correctly saved
6. Visit another users' profile, e.g. `/profile/madhavss`, `/profile/adviyer`, or `/profile/elliotk22`

Thanks! :smiley:
